### ISSUE TITLE
Data field on endpointConnect is not utilized.

### ISSUE THREAD

**Issue by dorflex150mg** [canonical/dqlite#287]
In dqlite/test/lib/server.c, the function endpointConnect takes a void pointer called data. It is not used in the function and takes NULL when called. Is this a future use or just a fault?

>    static int endpointConnect(void *data,
>    			   const char *address,
>    			   int *fd)
>    {    
>    	struct sockaddr_un addr;
>    	int rv;
>   	(void)address;
>   	(void)data;
>   	memset(&addr, 0, sizeof addr);
>    	addr.sun_family = AF_UNIX;
>    	strcpy(addr.sun_path + 1, address + 1);
>    	*fd = socket(AF_UNIX, SOCK_STREAM, 0);
>    	munit_assert_int(*fd, !=, -1);
>    	rv = connect(*fd, (struct sockaddr *)&addr, sizeof(sa_family_t) + strlen(address + 1) + 1);
>    	munit_assert_int(rv, ==, 0);
>    	return 0;
>    }

**Comment by freeekanayaka** [comment]
Probably it's just historical artifact and can be removed.

**Comment by shade34321** [comment]
I tried taking a stab at this but realized that it would need changes to `include/dqlite.h`, `src.server.c`, `src/transport.c` `src/transport.h` in addition to `test/lib/server.c`. I made several changes since it appeared to not be used but then ran across this and wasn't sure how to proceed.

https://github.com/canonical/dqlite/blob/master/src/transport.c#L73

Is it still unnecessary? If so any help would be appreciated in how to deal with it. Thanks!
